dspace-spring-rest/src/main/java/org/dspace/app/rest/MappingItemRestController.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.UUID;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.log4j.Logger;
+import org.dspace.app.rest.converter.ItemConverter;
+import org.dspace.app.rest.link.HalLinkService;
+import org.dspace.app.rest.model.ItemRest;
+import org.dspace.app.rest.model.MappingItemRestWrapper;
+import org.dspace.app.rest.model.hateoas.MappingItemResourceWrapper;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.app.rest.utils.Utils;
+import org.dspace.content.Collection;
+import org.dspace.content.Item;
+import org.dspace.content.service.CollectionService;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * This Controller will deal with request about the items that are mapped to the collection given in the request url
+ * Currently this only includes a GET method for all the items that are mapped to the given collection that do not
+ * have the given collection as their owning collection
+ */
+@RestController
+@RequestMapping("/api/core/collections/" +
+    "{uuid:[0-9a-fxA-FX]{8}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{12}}/mappedItems")
```

Comment:
@benbosman  and @Raf-atmire , I've just realized this endpoint isn't described in our REST Contract: https://github.com/DSpace/Rest7Contract/blob/master/collections.md

It's not a big deal for the purposes of approving this PR.  But, we need a separate PR for the REST Contract, please. 

For the REST Contract PR, we'll also need to ensure we make it clear the `/collections/[uuid]/mappedItems` endpoint is READ-ONLY.  Item mappings can only be modified via `/items/[uuid]/mappedCollections`.   I like that approach, we just need to document it.